### PR TITLE
Don't use Parity-specific JSONRPCs

### LINKED ISSUE #3: Make it work with MetaMask
Right now I have an error `Error: Pubsub is only available with a subscribing-supported transport injected!`.

### LINKED ISSUE #51: Light.js not working with Infura
Hey, been playing around with Light.js a bit, It's really awesome - really speeds up our DApp and simplifies the code! running into a problem with Infura though (MetaMask web3 provider is working well)

```js
web3jsInfura = new Web3(new Web3.providers.WebsocketProvider('wss://mainnet.infura.io/ws'));
light.setProvider(web3jsInfura.currentProvider);
```

`import { makeContract, withoutLoading } from "@parity/light.js";` the failures i'm seeing below are from reading state from a contract 

ie
```js
myContract.variable$()
          .pipe(
            withoutLoading()
          )
          .subscribe(r => {
            console.log('variable', r);
          });
```

screen shot 2018-11-22 at 10 46 39 am

**if i setup provider as in docs:**

`light.setProvider(
    Api.Provider.Ws('wss://mainnet.infura.io/ws')
);`

screen shot 2018-11-22 at 10 57 13 am

Thanks,
Mike

### PR DESCRIPTION
**Pull request by amaurymartiny**
Use
- `eth_subscribe('syncing')`
- `eth_subscribe('newHeads')`

instead of
- `parity_subscribe('eth_syncing')`
- `parity_subscribe('eth_blockNumber)`

In any case, if pubsub doesn't work, default to polling.

Notes: on parity-ethereum nodes, `eth_subscribe('syncing')` is not implemented, so it will poll.

Fixes #51, fixes #3

### CODE REVIEW

File: packages/api/src/pubsub/eth/eth.js [hunk 1]
```diff
@@ -15,213 +15,296 @@
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 const PubsubBase = require('../pubsubBase');
 
-const { inAddress, inBlockNumber, inHex, inNumber16, inOptions, inFilter } = require('../../format/input');
-const { outAddress, outBlock, outNumber, outTransaction, outSyncing, outReceipt, outLog } = require('../../format/output');
+const {
```
Comment (amaurymartiny) [inline packages/api/src/pubsub/eth/eth.js]: file diff mostly due to prettier :(

File: packages/light.js/src/frequency/blocks.ts [hunk 2]
```diff
@@ -20,11 +20,16 @@ import { onSyncingChanged$ } from './health';
...
 const onEveryBlockWithApi$ = memoizee(
   (api: any, options?: FrequencyObservableOptions) =>
-    createPubsubObservable('eth_blockNumber', options).pipe(
-      withLatestFrom(onSyncingChanged$(options)),
+    createPubsubObservable(
+      'eth_newHeads',
+      'eth_getBlockByNumber',
+      options
+    ).pipe(
+      withLatestFrom(onSyncingChanged$(options).pipe(startWith(false))),
```
Comment (axelchalon) [inline packages/light.js/src/frequency/blocks.ts]: I might be wrong but wouldn't this make onEveryBlockWithApi$ emit on initialization, even though the user is not synced? e.g.:
- onEveryBlockWithApi$ is called
- blockNumber RPC answers
- onEveryBlockWithApi$ emits wit the blockNumber
- isSyncing RPC answers `true` (which was always the case)
- onEveryBlockWithApi$ doesn't emit until isSyncing RPC answers `false`

File: packages/light.js/src/frequency/utils/createPubsubObservable.ts [hunk 3]
```diff
@@ -12,14 +12,36 @@ import { switchMap } from 'rxjs/operators';
...
+ */
+function createPoll<T> (
+  fallback: string,
+  api: any,
+  pollInterval = POLL_INTERVAL
+) {
+  const [fallbackNamespace, fallbackMethod] = fallback.split('_');
+
+  return timer(0, pollInterval).pipe(
+    switchMap(() => api[fallbackNamespace][fallbackMethod]())
```
Comment (axelchalon) [inline packages/light.js/src/frequency/utils/createPubsubObservable.ts]: might not be a problem but for RPC calls taking >1 second to answer, the observable would never fire

Comment (axelchalon) [reply packages/light.js/src/frequency/utils/createPubsubObservable.ts]: I think exhaustMap should do the trick https://runkit.com/axelchalon/5c00004a6d959e0012c60bc3 

We have a timer (source observable) ; we make the RPC query each time the timer fires, unless the previous RPC hasn't resolved yet

File: packages/light.js/src/frequency/blocks.ts (hunk 2 above)
Comment (axelchalon) [reply packages/light.js/src/frequency/blocks.ts]: I imagine a fix could be to make onSyncingChanged$ fire `eth_syncing` (non-pubsub rpc call) first to get an initial value, and then switch to the pubsub. non-pubsub "eth_syncing" does get an answer.

obviously not very clean though

Comment (amaurymartiny) [reply packages/light.js/src/frequency/blocks.ts]: I actually think it's a good idea, I thought about it too.

Another reason why it's good: `parity_subscribe('eth_getBlock')` immediately gives you an answer, and then gives you subsequent pubsub answers on each pub. `eth_subscribe('newHeads')` however doesn't give you the initial one, so there might be a (up to 15s) delay between subscribe and first event fired.

Comment (axelchalon) [reply packages/light.js/src/frequency/blocks.ts]: ah, makes sense then!

Comment (amaurymartiny) [reply packages/light.js/src/frequency/blocks.ts]: https://github.com/paritytech/js-libs/issues/61

**Pull request merged**